build-system/pr-check.js

```diff
@@ -0,0 +1,190 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview This file is executed by Travis (configured via
+ * .travis.yml in the root directory) and is the main driver script
+ * for running tests.  Execution herein is entirely synchronous, that
+ * is, commands are executed on after the other (see the exec
+ * function). Should a command fail, this script will then also fail.
+ * This script attempts to introduce some granularity for our
+ * presubmit checking, via the determineBuildTargets method.
+ */
+const child_process = require('child_process');
+const path = require('path');
+
+/**
+ * @param {string} str
+ * @param {string} prefix
+ * @return {boolean}
+ */
+function hasPrefix(str, prefix) {
+  return str.indexOf(prefix) == 0;
+}
+
+/**
+ * @param {string} str
+ * @param {string} prefix
+ * @return {boolean}
+ */
+function hasSuffix(str, suffix) {
+  return str.lastIndexOf(suffix) == str.length - suffix.length;
+}
+
+/**
+ * Executes the provided command, returning its stdout as an array of lines.
+ * This will throw an exception if something goes wrong.
+ * @param {string} cmd
+ * @return {!Array<string>}
+ */
+function exec(cmd) {
+  return child_process.execSync(cmd, {'encoding': 'utf-8'}).trim().split('\n');
+}
+
+/**
+ * Executes the provided command; terminates this program in case of failure.
+ * @param {string} cmd
+ */
+function execOrDie(cmd) {
+  console /*OK*/.log(`\npr-check.js: ${cmd}\n`);
```

Comment:
This came from the auto formatter I use (google-clang-format). I've whitelisted this file instead so this makes it more readable. And I did startsWith / endsWith as well while I was at it, since it's supported by Node.js v4.0.0 and later.